memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -119,24 +120,49 @@ object Histogram {
     final def numBuckets: Int = 0
     final def bucketTop(no: Int): Double = ???
     final def bucketValue(no: Int): Double = ???
-    final def serialize(intoBuf: Option[UnsafeBuffer] = None): UnsafeBuffer = {
+    final def serialize(intoBuf: Option[MutableDirectBuffer] = None): MutableDirectBuffer = {
       val buf = intoBuf.getOrElse(BinaryHistogram.histBuf)
       BinaryHistogram.writeNonIncreasing(HistogramBuckets.emptyBuckets, Array[Long](), buf)
       buf
     }
   }
 }
 
+trait HistogramWithBuckets extends Histogram {
+  def buckets: HistogramBuckets
+  final def numBuckets: Int = buckets.numBuckets
+  final def bucketTop(no: Int): Double = buckets.bucketTop(no)
+}
+
+final case class LongHistogram(buckets: HistogramBuckets, values: Array[Long]) extends HistogramWithBuckets {
```

Comment:
This PR too does not have the annotation on whether the histogram is based on counter based bucket values or increase based bucket values. This is important - I had called this out in earlier PR.

Reply:
Yeah, I still remember that.  It doesn't belong in each histogram though.  The counter vs increase is a property of the entire column, not each histogram.  I'll think about where to add that and the scope.